pkg/apiserver/controller/flavour/flavour.go

```diff
@@ -0,0 +1,201 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flavour
+
+import (
+	"reflect"
+
+	log "github.com/sirupsen/logrus"
+
+	"paddleflow/pkg/apiserver/common"
+	"paddleflow/pkg/apiserver/models"
+	"paddleflow/pkg/common/logger"
+	"paddleflow/pkg/common/schema"
+)
+
+// CreateFlavourRequest convey request for create flavour
+type CreateFlavourRequest struct {
+	Name            string                     `json:"name"`
+	ClusterName     string                     `json:"clusterName"`
+	ClusterID       string                     `json:"-"`
+	CPU             string                     `json:"cpu"`
+	Mem             string                     `json:"mem"`
+	ScalarResources schema.ScalarResourcesType `json:"scalarResources,omitempty"`
+	UserName        string                     `json:"-"`
+}
+
+// UpdateFlavourRequest convey request for update flavour
+type UpdateFlavourRequest struct {
+	Name            string                     `json:"-"`
+	ClusterName     string                     `json:"clusterName,omitempty"`
+	ClusterID       string                     `json:"-"`
+	CPU             string                     `json:"cpu,omitempty"`
+	Mem             string                     `json:"mem,omitempty"`
+	ScalarResources schema.ScalarResourcesType `json:"scalarResources,omitempty"`
+	UserName        string                     `json:"-"`
+}
+
+// CreateFlavourResponse convey response for create flavour
+type CreateFlavourResponse struct {
+	models.Flavour
+}
+
+// UpdateFlavourResponse convey response for update flavour
+type UpdateFlavourResponse struct {
+	models.Flavour
+}
+
+// ListFlavourResponse convey response for list flavour
+type ListFlavourResponse struct {
+	common.MarkerInfo
+	FlavourList []models.Flavour `json:"flavourList"`
+}
+
+// CreateFlavour handler for creating flavour
+func CreateFlavour(request *CreateFlavourRequest) (*CreateFlavourResponse, error) {
+	flavour := models.Flavour{
+		Name:            request.Name,
+		CPU:             request.CPU,
+		Mem:             request.Mem,
+		ScalarResources: request.ScalarResources,
+		ClusterID:       request.ClusterID,
+		ClusterName:     request.ClusterName,
+		UserName:        request.UserName,
+	}
+	if err := models.CreateFlavour(&flavour); err != nil {
+		return nil, err
+	}
+
+	response := &CreateFlavourResponse{
+		Flavour: flavour,
+	}
+	return response, nil
+}
+
+// UpdateFlavour handler for updating flavour
+func UpdateFlavour(request *UpdateFlavourRequest) (*UpdateFlavourResponse, error) {
+	flavour, err := GetFlavour(request.Name)
+	if err != nil {
+		log.Errorf("get flavour %s failed when update", request.Name)
+		return nil, err
+	}
+	isChanged := false
+	if request.CPU != flavour.CPU {
+		isChanged = true
+		flavour.CPU = request.CPU
```

Comment:
Check the value of parameter, it might be better to check at router/v1/flavour file.